packages/world-modules/src/modules/puppet/createPuppet.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { IBaseWorld } from "@latticexyz/world/src/codegen/interfaces/IBaseWorld.sol";
+import { WorldResourceIdInstance } from "@latticexyz/world/src/WorldResourceId.sol";
+import { PUPPET_DELEGATION, PUPPET_FACTORY } from "./constants.sol";
+import { PuppetDelegationControl } from "./PuppetDelegationControl.sol";
+import { Puppet } from "./Puppet.sol";
+import { PuppetFactorySystem } from "./PuppetFactorySystem.sol";
+
+using WorldResourceIdInstance for ResourceId;
+
+/**
+ * This free function can be used to create a puppet and register it with the puppet delegation control.
+ * Since it is inlined in the caller's context, the calls originate from the caller's address.
+ */
+function createPuppet(IBaseWorld world, ResourceId systemId) returns (address puppet) {
+  puppet = abi.decode(
+    world.call(PUPPET_FACTORY, abi.encodeCall(PuppetFactorySystem.createPuppet, (systemId))),
```

Comment:
ooc why don't we use SystemSwitch here?

Reply:
Agree should probably use system switch here

Reply:
update: turns out using system switch here pushes `ERC20Module` over the bytecode limit, so reverting for now (meaning you can't register a puppet from a root system for now). We should find a better implementation for `SystemSwitch` that doesn't require inlining the libraries, maybe by deploying `StoreCore` as a public library.

Reply:
we should prob add a follow up issue to come back to 1) this specific instance and 2) the problem overall

Reply:
yeah i mean public as in not-internal, which means delegatecall to an external contract